File: Python/Problem_010.py
# Find the sum of all the primes below two million.

limit = 2000000

sum = 0
non_primes = set() # set of all non prime numbers
for i in range(2, limit):
    # Add to sum if prime
    if i not in non_primes:
        sum += i
    # Add all multiples of i below limit to non_primes
    non_primes.update(range(i*2, limit, i))

print("The sum of all the primes below 2 million is %d" % sum)

File: C#/Problem_010.cs
using System;

namespace EulerProject
{
    class Problem_10
    {
        static void main(string[] args)
        {
            // Find the sum of all the primes below two million.

            int limit = 2000000;
            long sum = 2;
            for (int j = 3; j < limit; j+=2)
            {
                if (isPrime(j))
                {
                    sum += j;
                }
            }
            System.Console.WriteLine($"The sum of all the primes below two million is {sum}.");
        }

        static bool isPrime(int x)
        {
            for (int i = 2; i < Math.Sqrt(x) + 1; i++)
            {
                if (x % i == 0)
                {
                    return false;
                }
            }
            return true;
        }
    }
}